microsoft/testsuites/libvirt/libvirt_tck_tool.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import xml.etree.ElementTree as ETree
+from dataclasses import dataclass
+from pathlib import Path, PurePath
+from typing import Any, List, Type, cast
+
+from assertpy.assertpy import assert_that
+
+from lisa import Environment, notifier
+from lisa.executable import Tool
+from lisa.messages import CommunityTestMessage, TestStatus, create_test_result_message
+from lisa.operating_system import Posix
+from lisa.testsuite import TestResult
+from lisa.tools import Git
+
+
+@dataclass
+class LibvirtTckTestResult:
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+
+
+class LibvirtTck(Tool):
+    TIME_OUT = 3600
+
+    # The failures in these tests need to be investigated and fixed. Until then, treat
+    # them as expected failures.
+    EXPECTED_FAILURES = [
+        "100-apply-verify-host_t",
+        "220-no-ip-spoofing_t",
+        "230-no-mac-broadcast_t",
+    ]
+
+    repo = "https://gitlab.com/libvirt/libvirt-tck.git"
+    deps = [
+        "expat-devel",
+        "gcc",
+        "libguestfs-tools",
+        "libsys-virt-perl",
+        "libtest-xml-perl",
+        "libvirt-daemon-system",
+        "libvirt-dev",
+        "make",
+        "perl",
+        "qemu-kvm",
+        "qemu-system-x86",
+    ]
+
+    repo_root: PurePath
+
+    @property
+    def command(self) -> str:
+        return "libvirt-tck"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git]
+
+    def run_tests(
+        self,
+        test_result: TestResult,
+        environment: Environment,
+        log_path: Path,
+    ) -> None:
+        result = self.run(
+            "--force --format junit -a results.tar.gz",
+            timeout=self.TIME_OUT,
+            force_run=True,
+            cwd=self.repo_root,
+            sudo=True,
+            shell=True,
+        )
+
+        results = self._extract_test_results(result.stdout)
+        failures = [r.name for r in results if r.status == TestStatus.FAILED]
+        expected_fails = [r.name for r in results if r.status == TestStatus.ATTEMPTED]
+        if not failures and not expected_fails:
+            result.assert_exit_code()
+
+        for r in results:
+            self._send_community_test_msg(
+                test_result.id_,
+                environment,
+                r.name,
+                r.status,
+            )
+
+        self.node.shell.copy_back(
+            self.repo_root / "results.tar.gz",
+            log_path / "libvirt_tck_results.tar.gz",
+        )
+        assert_that(failures, f"Unexpected failures: {failures}").is_empty()
```

Comment:
clean up results.tar.gz file

Reply:
Why? Test VMs will be destroyed anyway.

Reply:
probably this vm will be reused for the same or different cases

Reply:
If it is reused for same case, then the file will simply get overwritten. If it is reused for different case this file will not affect it in any way.